frontend/src/profile/GraduationShareDialog.tsx

```diff
@@ -0,0 +1,168 @@
+        // https://github.com/bubkoo/html-to-image/issues/40
+        // https://stackoverflow.com/questions/42263223/how-do-i-handle-cors-with-html2canvas-and-aws-s3-images
+        // https://www.hacksoft.io/blog/handle-images-cors-error-in-chrome
+        domToPng(node, {
+            debug: true,
+            fetch: {
+                bypassingCache: true,
+            },
+            quality: 1,
+            scale: 2,
+            backgroundColor: '#121212',
+        })
+            .then((dataUrl) => {
+                setImageData(dataUrl);
+            })
+            .catch((error) => {
+                console.error('domToPng: ', error);
+            });
+    };
+
+    useEffect(() => {
+        renderImage();
+    });
+
+    const onDownload = (closeAfter: boolean) => {
+        if (!imageData) {
+            return;
+        }
+        trackEvent(EventType.DownloadGradBox, {
+            previous_cohort: graduation.previousCohort,
+            new_cohort: graduation.newCohort,
+            dojo_score: graduation.score,
+            graduated_at: graduation.createdAt,
+        });
+
+        const link = document.createElement('a');
+        link.href = imageData;
+        link.download = `graduation-${newCohort}.png`;
+        document.body.appendChild(link);
+        link.click();
+        document.body.removeChild(link);
+        if (closeAfter) {
+            onClose();
+        }
+    };
+
+    return (
+        <Dialog maxWidth='md' open={open} onClose={onClose} fullWidth>
+            <DialogTitle>Share your progress!</DialogTitle>
+            <DialogContent>
+                <Stack spacing={2}>
+                    <DialogContentText>
+                        Show off your hard work and welcome the world to the Dojo!
+                        Download this image and share on social media.
+                    </DialogContentText>
+                    <Stack
+                        display='grid'
+                        gridTemplateRows='auto 1fr'
+                        alignItems='center'
+                        justifyContent='center'
+                    >
+                        {imageData ? (
+                            <img
+                                onClick={() => onDownload(false)}
+                                style={{ height: 'auto', maxWidth: '100%' }}
+                                alt='dojo graduation badge'
+                                src={imageData}
+                            />
+                        ) : (
+                            <ReportCanvas ref={reportRef}>
+                                <GraduationCard graduation={graduation} />
+                            </ReportCanvas>
+                        )}
+                    </Stack>
+                </Stack>
+            </DialogContent>
+            <DialogActions>
+                <Button onClick={() => onClose()}>Close</Button>
+                <LoadingButton loading={!imageData} onClick={() => onDownload(true)}>
```

Comment:
I don't think the download button should close the dialog either. That surprised me when I was testing it earlier

Reply:
Okay, good to know it surprised you - for sure that means we should change it.

Reply:
Ah. I see you already made the change.